src/app/features/notifications/notifications.component.ts

```diff
@@ -94,18 +85,51 @@ export class NotificationsComponent implements OnInit {
   private filterUserNotifications(
     notifications: NotificationActivity[],
     user: string,
-  ): NotificationActivity[] {
-    return notifications.filter((notification) => notification.sentBy === user);
+  ) {
+    this.allNotifications = notifications.filter(
+      (notification) => notification.sentBy === user,
+    );
+    this.unreadNotifications = notifications.filter(
+      (notification) =>
+        notification.sentBy === user && notification.readStatus === false,
+    );
   }
 
   /**
    * Counts unread notifications from the list.
    * @param notifications - The list of notifications.
    */
-  private countUnreadNotifications(
-    notifications: NotificationActivity[],
-  ): number {
-    return notifications.filter((notification) => !notification.readStatus)
-      .length;
+  private countUnreadNotifications(notifications: NotificationActivity[]) {
+    return notifications.filter(
+      (notification) => notification.readStatus === false,
+    ).length;
+  }
+
+  markAllRead($event: Event) {
+    $event.stopPropagation();
+    Logging.log("All notifications marked as read");
+  }
```

Comment:
Need to update the logic so that when the user click on the mark all as read button then all the notification mark as read.

Reply:
Can you implement that now already to finalize this component? (I guess updating all unread notifications and then saving those using the entityMapper?)

Reply:
For these types of functionalities, I have created this [issue](https://github.com/Aam-Digital/ndb-core/issues/2690). Can we implement this functionality under that issue?